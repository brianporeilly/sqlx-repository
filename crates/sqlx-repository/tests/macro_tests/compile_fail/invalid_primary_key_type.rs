//! Test that repository derive fails with non-i32 primary key

use sqlx_repository::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow, Repository)]
#[repository(table = "users")]
pub struct User {
    pub id: String,  // Primary key must be i32 for now
    pub name: String,
    pub email: String,
}

fn main() {}