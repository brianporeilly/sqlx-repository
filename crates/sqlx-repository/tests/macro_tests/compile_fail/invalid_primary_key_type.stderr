error: Primary key 'id' must be of type 'i32', found 'String'.

       Currently supported: i32
       Future versions will support: i64, Uuid

       Example:
       pub struct User {
           pub id: i32,  // ← Must be i32
           pub name: String,
       }
 --> tests/macro_tests/compile_fail/invalid_primary_key_type.rs:9:13
  |
9 |     pub id: String,  // Primary key must be i32 for now
  |             ^^^^^^
